media-proxy/src/session-base.cc

```diff
@@ -0,0 +1,211 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <string.h>
+#include <strings.h>
+#include <sys/stat.h>
+#include <arpa/inet.h>
+
+#include <mcm_dp.h>
+
+#include "session-base.h"
+#include <unistd.h>
+
+static int on_connect_callback_wrapper(memif_conn_handle_t conn, void *priv_data)
+{
+    session *connection = (session *)priv_data;
+    return connection->on_connect_cb(conn);
+}
+
+static int on_disconnect_callback_wrapper(memif_conn_handle_t conn, void *priv_data)
+{
+    session *connection = (session *)priv_data;
+    return connection->on_disconnect_cb(conn);
+}
+
+static int on_receive_callback_wrapper(memif_conn_handle_t conn, void *priv_data, uint16_t qid)
+{
+    session *connection = (session *)priv_data;
+    return connection->on_receive_cb(conn, qid);
+}
+
+void session::memif_event_loop()
+{
+    int err;
+    memif_socket_handle_t memif_socket = memif_conn_args.socket;
+
+    do {
+        // INFO("media-proxy waiting event.");
+        err = memif_poll_event(memif_socket, -1);
+        // INFO("media-proxy received event.");
+    } while (err == MEMIF_ERR_SUCCESS);
+
+    INFO("MEMIF DISCONNECTED.");
+}
+
+int session::shm_init(memif_ops_t *memif_ops, uint32_t buffer_size, uint32_t log2_ring_size)
+{
+    int ret = 0;
+
+    /* Set application name */
+    strncpy(memif_socket_args.app_name, memif_ops->app_name,
```

Comment:
```suggestion
    strlcpy(memif_socket_args.app_name, memif_ops->app_name,
```
to be sure that the string is NULL terminated. Should be done everywhere.

Reply:
Agree. But when changing `strncpy` to `strlcpy` please note you don't need to subtract 1 from the size.

Reply:
done